docs/src/pages/components/data-grid/rendering/CustomPaginationGrid.tsx

```diff
@@ -10,8 +10,8 @@ const useStyles = makeStyles({
   },
 });
 
-function CustomPagination(props: GridBaseComponentProps) {
-  const { state, api } = props;
+function CustomPagination() {
+  const { state, apiRef } = useGridBaseComponentProps();
```

Comment:
how about we remove `Base` just call it `useGridComponentProps`?

Reply:
If we wrote this in the documentation, would it be clear enough?

> To access the component props use: `useGridComponentProps`

Reply:
Same. If I read `useGridComponentProps` I will probably make an assumption that this will get me the props on the actual grid component, not a subset that is passed to a slot component. I would even argue that a better name would be `useGridSlotComponentProps`.

Reply:
Renamed